### PR TITLE
Add check-manifest-yaml.rb

### PR DESCRIPTION
**Pull request by tk0miya**
Add `check-manifest-yaml.rb` to check manifest.yaml file on each gems. It verifies

* manifest.yaml contains a "dependencies" field
* each dependency contains a "name" field
* each dependency name points a stdlib

refs: #267

### CODE REVIEW

**Review by soutaro — APPROVED** [review]
Thanks! 🎉

**Review by ksss — APPROVED** [review]
👍

File: gems/railties/6.0/_scripts/test [hunk 1]
```diff
@@ -19,3 +19,4 @@ bundle exec rbs --repo $REPO_DIR -r tsort validate --silent
 # bundle exec steep check
 #
 # $(git rev-parse --show-toplevel)/bin/check-untyped-call.rb
+$(git rev-parse --show-toplevel)/bin/check-manifest-yaml.rb
```
Comment (tk0miya) [inline gems/railties/6.0/_scripts/test:22]: I added this line after posting this PR. This file was added at https://github.com/ruby/gem_rbs_collection/pull/417

**Review by yykamei — APPROVED** [review]
Thank you!

File: bin/check-manifest-yaml.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    exit 1
+  end
+
+  manifest['dependencies'].each do |dep|
+    unless dep.key?('name')
+      puts "#{dep} does not contain the name of dependency."
+      exit 1
+    end
+
+    unless stdlib?(dep['name'])
```
Comment (pocke) [inline bin/check-manifest-yaml.rb:33]: This check will be unnecessary. I'll remove it after the bug is fixed. 
https://github.com/ruby/rbs/issues/1538

**Review by pocke — APPROVED** [review]

**Pull request merged**